go/pkg/casng/uploader.go

```diff
@@ -343,5 +424,40 @@ func (u *uploader) close() {
 	log.V(1).Infof("[casng] uploader: waiting for workers")
 	u.workerWg.Wait()
 
+	close(u.logBeatDoneCh)
 	log.V(3).Infof("[casng] upload.close.duration: start=%d, end=%d", startTime.UnixNano(), time.Now().UnixNano())
 }
+
+func (u *uploader) logBeat() {
+	var interval time.Duration
+	if log.V(3) {
+		interval = time.Second
+	} else if log.V(2) {
+		interval = 30 * time.Second
+	} else if log.V(1) {
+		interval = time.Minute
+	} else {
+		return
+	}
+
+	log.Infof("[casng] beat.start; interval=%v", interval)
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	i := 0
+	for {
+		select {
+		case <-u.logBeatDoneCh:
+			log.Infof("[casng] beat.stop; interval=%v, count=%d", interval, i)
+			return
+		case <-ticker.C:
+		}
+
+		i++
+		log.Infof("[casng] beat; #%d, upload_subs=%d, query_subs=%d, walkers=%d, batching=%d, streaming=%d, querying=%d, open_files=%d, large_open_files=%d",
+			i, u.uploadPubSub.len(), u.queryPubSub.len(), u.walkThrottler.len(), u.uploadThrottler.len(), u.streamThrottle.len(), u.queryThrottler.len(), u.ioThrottler.len(), u.ioLargeThrottler.len())
+	}
+}
+
+func isExec(mode fs.FileMode) bool {
+	return mode&0100 != 0
```

Comment:
Does this work on windows?

Reply:
Probably not. However, if the upstream call is coming from the server, it should work since the bits are constructed from a boolean. If the upstream call is coming from stating the filesystem, the value doesn't matter because a subsequent client cannot (or shouldn't) make use of this bit because Windows doesn't have an executable bit.